src/modules/fw_pos_control_l1/fw_pos_control_l1_main.cpp

```diff
@@ -862,6 +904,31 @@ void FixedwingPositionControl::navigation_capabilities_publish()
 	}
 }
 
+void FixedwingPositionControl::get_waypoint_heading_distance(float heading, float distance,
+					struct position_setpoint_s &waypoint_prev, struct position_setpoint_s &waypoint_next, bool flag_init)
+{
+	waypoint_prev.valid = true;
+	waypoint_prev.alt = _hold_alt;
+
+	if (flag_init) {
+		// on init set first waypoint to momentary position
+		waypoint_prev.lat = _global_pos.lat;
+		waypoint_prev.lon = _global_pos.lon;
+	} else {
+		/*
```

Comment:
@tumbili @AndreasAntener @kd0aij This else case is the issue. Both waypoints have to evaluated fresh here, one put  behind the plane and one in front. The way its done right now is not a complete transformation.